Code/Framework/AzToolsFramework/AzToolsFramework/AssetBrowser/Entries/RootAssetBrowserEntry.cpp

```diff
@@ -127,6 +130,33 @@ namespace AzToolsFramework
                 source->m_displayName = QString::fromUtf8(source->m_name.c_str());
                 source->m_scanFolderId = fileDatabaseEntry.m_scanFolderPK;
                 source->m_extension = absoluteFilePath.Extension().Native();
+                source->m_diskSize = AZ::IO::SystemFile::Length(absoluteFilePath.c_str());
+                AZ::IO::FixedMaxPath assetPath;
+                if (auto settingsRegistry = AZ::SettingsRegistry::Get(); settingsRegistry != nullptr)
+                {
+                    settingsRegistry->Get(assetPath.Native(), AZ::SettingsRegistryMergeUtils::FilePathKey_CacheRootFolder);
+                    assetPath /= fileDatabaseEntry.m_fileName + ".metadata.json";
+                    FILE* f1;
+                    azfopen(&f1, assetPath.c_str(), "r");
+                    if (f1)
+                    {
+                        char readBuffer[65536];
+                        rapidjson::FileReadStream is(f1, readBuffer, sizeof(readBuffer));
+
+                        rapidjson::Document doc;
+                        doc.ParseStream(is);
+                        fclose(f1);
```

Comment:
Replace with one of the JSON reading helpers, such as `JsonSerializationUtils::ReadJsonFile`

Reply:
Done.